frontend/actions/src/main/java/com/virtuslab/gitmachete/frontend/actions/traverse/TraverseSyncToParent.java

```diff
@@ -0,0 +1,133 @@
+      INonRootManagedBranchSnapshot gitMacheteBranch,
+      @UI Runnable traverseNextEntry) {
+    this.project = gitRepository.getProject();
+    this.gitRepository = gitRepository;
+    this.graphTable = graphTable;
+    this.repositorySnapshot = repositorySnapshot;
+    this.gitMacheteBranch = gitMacheteBranch;
+    this.traverseNextEntry = traverseNextEntry;
+  }
+
+  @UIEffect
+  public void sync() {
+    @UI Runnable syncToRemoteRunnable = new TraverseSyncToRemote(gitRepository, graphTable, gitMacheteBranch,
+        traverseNextEntry)::sync;
+
+    switch (gitMacheteBranch.getSyncToParentStatus()) {
+      case MergedToParent :
+        if (!handleMergedToParent()) {
+          return;
+        }
+        break;
+
+      case InSyncButForkPointOff :
+      case OutOfSync :
+        if (!handleOutOfSyncOrInSyncButForkPointOff(syncToRemoteRunnable)) {
+          return;
+        }
+        break;
+
+      default :
+        break;
+    }
+
+    graphTable.queueRepositoryUpdateAndModelRefresh(syncToRemoteRunnable);
+  }
+
+  @UIEffect
+  private boolean handleMergedToParent() {
+    val branchLayout = repositorySnapshot.getBranchLayout();
+    val currentBranchIfManaged = repositorySnapshot.getCurrentBranchIfManaged();
+    val slideOutDialog = MessageDialogBuilder.yesNoCancel(
+        getString("action.GitMachete.BaseTraverseAction.dialog.merged-to-parent.title"),
+        getString(
+            "action.GitMachete.BaseTraverseAction.dialog.merged-to-parent.text.HTML").fmt(
+                gitMacheteBranch.getName()))
+        .cancelText(getString("action.GitMachete.BaseTraverseAction.dialog.cancel-traverse"));
+
+    switch (slideOutDialog.show(project)) {
+      case MessageConstants.YES :
+        new SlideOutBackgroundable(getString("action.GitMachete.BaseSlideOutAction.task.title"),
+            gitMacheteBranch, gitRepository, currentBranchIfManaged, branchLayout, graphTable) {
+          @Override
+          public void onSuccess() {
+            graphTable.queueRepositoryUpdateAndModelRefresh(traverseNextEntry);
+          }
+        }.queue();
+        return false;
+
+      case MessageConstants.NO :
+        break;
+
+      default :
+        return false;
+    }
+    return true;
+  }
+
+  @UIEffect
+  private boolean handleOutOfSyncOrInSyncButForkPointOff(@UI Runnable syncToRemoteRunnable) {
+    val rebaseDialog = MessageDialogBuilder.yesNoCancel(
+        getString("action.GitMachete.BaseTraverseAction.dialog.diverged-from-parent.title"),
+        getString(
+            "action.GitMachete.BaseTraverseAction.dialog.diverged-from-parent.text.HTML").fmt(
+                gitMacheteBranch.getName(),
+                gitMacheteBranch.getParent().getName()))
+        .cancelText(getString("action.GitMachete.BaseTraverseAction.dialog.cancel-traverse"));
+
+    switch (rebaseDialog.show(project)) {
+      case MessageConstants.YES :
+        new RebaseOnParentBackgroundable(
```

Comment:
Whooops I'm getting such a small dialog:

<img width="509" alt="image" src="https://user-images.githubusercontent.com/3383210/205659215-3b54267a-27a3-4c20-b1fe-ca5e0c2348a2.png">

Reply:
Although huh, it also happens when running rebase from outside of traverse hmmm

Reply:
happens to me sometimes with different dialogs. I think it's  some issue with IntelliJ itself